JsonSchema/RefKeyword.cs

```diff
@@ -49,12 +49,25 @@ public void Evaluate(EvaluationContext context)
 
 		Uri newUri;
 		string fragment;
-
 		// If the uri is a file we need to set the fragment manually because it will be lost in the uri
-		if (context.Scope.LocalScope.IsFile && Reference.OriginalString.StartsWith("#"))
+		if (context.Scope.LocalScope.IsFile && Reference.OriginalString.Contains("#"))
 		{
-			newUri = context.Scope.LocalScope;
-			fragment = Reference.OriginalString;
+			if (Reference.OriginalString.StartsWith("#"))
+			{
+				fragment = Reference.OriginalString;
+				newUri = context.Scope.LocalScope;
+			}
+
+			else
+			{
+				var parts=Reference.OriginalString.Split('#');
+				if (parts.Length != 2)
+					throw new JsonSchemaException(
+						$"Given a reference with more than one '#' in it. We cannot tell if this is a fragment, or where the fragment starts. Please don't use '#'s in fileNames or paths.\n Reference:{Reference.OriginalString} ");
```

Comment:
This is rather long.  I'd prefer a shorter error message with a longer explanation in the docs.

```suggestion
						"File references with multiple hashes are not supported.");
```